components/paging/pageable-subscriber-mixin.js

```diff
@@ -0,0 +1,33 @@
+import { EventSubscriberController, IdSubscriberController } from '../../controllers/subscriber/subscriberControllers.js';
+
+export const PageableSubscriberMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			/**
+			 * Id of the `PageableMixin` component this component wants to observe (if not located within that component)
+			 * @type {string}
+			 */
+			pageableFor: { type: String, reflect: true, attribute: 'pageable-for' },
+			_pageableInfo: { state: true }
+		};
+	}
+
+	constructor() {
+		super();
+
+		this._pageableInfo = { itemCount: -1, itemShowingCount: 0 };
+		this._pageableEventSubscriber = new EventSubscriberController(this, 'pageable');
+		this._pageableIdSubscriber = new IdSubscriberController(this, 'pageable', { idPropertyName: 'pageableFor' });
+	}
+
+	async getUpdateComplete() {
+		await super.getUpdateComplete();
+		await (this.pageableFor ? this._pageableIdSubscriber._updateComplete : this._pageableEventSubscriber._updateComplete);
+	}
```

Comment:
A bit of magic to make the tests work nicely without having to add an explicit `requestAnimationFrame` in them. Should also benefit anything else that's waiting on the subscriber.

Reply:
What does `_updateComplete` represent here?

Reply:
The pageable subscriber would also have `updateComplete`. I am wondering if we should name this differently to avoid any confusion. Thoughts?

Reply:
Are you thinking renaming the subscriber controllers, like:

```js
await (this.pageableFor ? this._pageableIdSubscriberController._updateComplete : this._pageableEventSubscriberController._updateComplete);
```

Reply:
Another option to consider is renaming this whole mixin as `PageableObserverMixin`, which would be consistent with `SelectionObserverMixin`. Then "subscriber" refers to the controller and "observer" refers to the mixin/component?